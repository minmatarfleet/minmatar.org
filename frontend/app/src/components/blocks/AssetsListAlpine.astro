---
import { i18n } from '@helpers/i18n'
const { t } = i18n(Astro.url)

import AssetsBlockAlpine from '@components/blocks/AssetsBlockAlpine.astro';
import TextBox from '@components/layout/TextBox.astro';
import Flexblock from '@components/compositions/Flexblock.astro';
---

<Flexblock
    id="assets-list"
    class="[ assets-list ]"
    {...Astro.props}
>
    <template x-for="assets_location in assets_locations_filtered">
        <AssetsBlockAlpine />
    </template>
    <template x-if="assets_locations_filtered.length == 0">
        {!Astro.slots.has('default') &&
            <TextBox>
                <p>{t('no_assets')}</p>
            </TextBox>
        }
    </template>
    <slot />
</Flexblock>

<style lang="scss">
    .assets-list {
        position: relative;
        //margin-top:
        top: calc(-1 * var(--space-s));
    }
</style>