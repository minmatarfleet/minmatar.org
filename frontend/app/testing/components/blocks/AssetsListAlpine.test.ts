import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import AssetsListAlpine from "@components/blocks/AssetsListAlpine.astro";

test("AssetsListAlpine defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(AssetsListAlpine, {});

  expect(result).toMatchSnapshot();
});
