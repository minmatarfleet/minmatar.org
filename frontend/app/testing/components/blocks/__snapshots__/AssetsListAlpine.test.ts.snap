// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AssetsListAlpine defaults 1`] = `
"<div class="flex-block [ assets-list ] gap---space-s" id="assets-list" data-astro-cid-2qz5zwg2="true" data-mod style="--flex_block_gap: var(--space-s);">  <template x-for="assets_location in assets_locations_filtered" data-astro-cid-2qz5zwg2> <div class="assets-location-list" data-astro-cid-2qz5zwg2="true" data-astro-cid-hpbvoymk> <div class="flex-block gap-0" x-data="{
        expanded: true,
        location_assets_count: 0,
        collapse() {
            this.expanded = false
        },
        toggle_collapse() {
            this.expanded = !this.expanded
            global_collapse = false
        },
    }" x-effect="global_collapse &#38;&#38; collapse()" data-astro-cid-xzc67zbc="true" data-mod style="--flex_block_gap: 0;">  <div class="[ sticky-header ]" x-bind:style="\`top: \${post_header_height - 2}px;\`" data-astro-cid-xzc67zbc> <div class="wrapper [ card component ] [ border-bottom ] light-transparency" data-astro-cid-xzc67zbc="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-l-xl);--wrapper_padding_inline_end: var(--space-l-xl);--wrapper_padding_block: var(--component-padding-block);--wrapper_max_width: var(--max-text-width);">   <button class="[ !block !w-full ] astro-muoml3se" x-on:click="toggle_collapse()" data-astro-cid-xzc67zbc="true" type="button" data-astro-cid-muoml3se> <div class="fluid-fixed [ items-center ]" data-astro-cid-muoml3se="true" data-mod data-astro-cid-rwsocfon style="--fixed_fluid_gap: var(--space-l);--fixed_fluid_width: 20px;--fixed_fluid_breakpoint: 50%;">   <div class="flex-block [ asset-location-item ] [ w-full ] gap-0" slot="header" data-astro-cid-hpbvoymk="true" x-data="{
        get_assets_count() {
            return assets_location.assets.reduce( (c, i) => c + (i.count ?? 1) , 0 )
        }
    }" data-mod style="--flex_block_gap: 0;">  <span class="text-[var(--highlight)]" x-text="assets_location.location_name"></span> <small x-text="\`\${get_assets_count()} \${get_assets_count() != 1 ? 'assets' : 'asset'} in inventory\`"></small>  </div>   <svg width="12" height="8" viewBox="0 0 12 8" fill="none" :class="expanded ? 'expanded' : 'collapsed'" data-astro-cid-muoml3se="true"> <path d="M10.59 0.000156279L6 4.58016L1.41 0.000155477L5.24537e-07 1.41016L6 7.41016L12 1.41016L10.59 0.000156279Z" fill="currentColor"></path> </svg>  </div>   </button>    </div>   </div> <div class="wrapper [ card component ] light-transparency" x-show="expanded" x-collapse.duration.0ms="true" data-astro-cid-xzc67zbc="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-l-xl);--wrapper_padding_inline_end: var(--space-l-xl);--wrapper_padding_block: var(--component-padding-block);--wrapper_max_width: var(--max-text-width);">     <template x-if="assets_style === 'list'" data-astro-cid-hpbvoymk> <div class="flex-block [ assets-items ] [ min-h-[64px] justify-center ] gap---space-3xs" data-astro-cid-hpbvoymk="true" data-mod style="--flex_block_gap: var(--space-3xs);">  <template x-for="asset in assets_location.assets" data-astro-cid-hpbvoymk> <div data-astro-cid-ekkdkgyu> <template x-if="assets_style === 'list'" data-astro-cid-ekkdkgyu> <div class="fixed-fluid [ asset-item ]" data-astro-cid-hpbvoymk="true" data-astro-cid-ekkdkgyu="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 64px;--fixed_fluid_breakpoint: 50%;">  <picture data-astro-cid-ekkdkgyu> <img loading="lazy" x-bind:src="get_item_icon(asset.id)" width="64" height="64" x-bind:alt="\`\${asset.name} icon\`" data-astro-cid-ekkdkgyu> </picture> <div class="flex-inline" data-astro-cid-ekkdkgyu="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-s);--flex_inline_justification: space-between;">  <div class="!flex !items-center" data-astro-cid-ekkdkgyu="true">  <div class="flex-block [ w-full ] gap-0" data-astro-cid-ekkdkgyu="true" data-mod style="--flex_block_gap: 0;">  <span x-text="\`\${asset.count} x \${asset.name}\`" data-astro-cid-ekkdkgyu></span>  </div>   </div>  </div>    </div>   </template> <template x-if="assets_style === 'icons'" data-astro-cid-ekkdkgyu> <div class="flex-block [ asset-icon ] gap---space-3xs" data-astro-cid-ekkdkgyu="true" data-mod="centered" style="--flex_block_gap: var(--space-3xs);">  <picture data-astro-cid-ekkdkgyu> <img loading="lazy" x-bind:src="get_item_icon(asset.id)" width="64" height="64" x-bind:alt="\`\${asset.name} icon\`" data-astro-cid-ekkdkgyu> </picture> <small x-text="\`\${asset.name}\`" data-astro-cid-ekkdkgyu></small>  </div>  </template> </div>  </template> <template x-if="assets_location.assets.length == 0" data-astro-cid-hpbvoymk> No assets found at location </template>  </div>  </template> <template x-if="assets_style === 'icons'" data-astro-cid-hpbvoymk> <div class="grid grid-fill" data-astro-cid-hpbvoymk="true" data-astro-cid-fkxmsq5x style="--grid_row_gap: var(--space-3xs);--grid_column_gap: var(--space-2xs);--grid_min_item_width: 64px;">  <template x-for="asset in assets_location.assets" data-astro-cid-hpbvoymk> <div data-astro-cid-ekkdkgyu> <template x-if="assets_style === 'list'" data-astro-cid-ekkdkgyu> <div class="fixed-fluid [ asset-item ]" data-astro-cid-hpbvoymk="true" data-astro-cid-ekkdkgyu="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 64px;--fixed_fluid_breakpoint: 50%;">  <picture data-astro-cid-ekkdkgyu> <img loading="lazy" x-bind:src="get_item_icon(asset.id)" width="64" height="64" x-bind:alt="\`\${asset.name} icon\`" data-astro-cid-ekkdkgyu> </picture> <div class="flex-inline" data-astro-cid-ekkdkgyu="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-s);--flex_inline_justification: space-between;">  <div class="!flex !items-center" data-astro-cid-ekkdkgyu="true">  <div class="flex-block [ w-full ] gap-0" data-astro-cid-ekkdkgyu="true" data-mod style="--flex_block_gap: 0;">  <span x-text="\`\${asset.count} x \${asset.name}\`" data-astro-cid-ekkdkgyu></span>  </div>   </div>  </div>    </div>   </template> <template x-if="assets_style === 'icons'" data-astro-cid-ekkdkgyu> <div class="flex-block [ asset-icon ] gap---space-3xs" data-astro-cid-ekkdkgyu="true" data-mod="centered" style="--flex_block_gap: var(--space-3xs);">  <picture data-astro-cid-ekkdkgyu> <img loading="lazy" x-bind:src="get_item_icon(asset.id)" width="64" height="64" x-bind:alt="\`\${asset.name} icon\`" data-astro-cid-ekkdkgyu> </picture> <small x-text="\`\${asset.name}\`" data-astro-cid-ekkdkgyu></small>  </div>  </template> </div>  </template> <template x-if="assets_location.assets.length == 0" data-astro-cid-hpbvoymk> No assets found at location </template>  </div>   </template>    </div>    </div>   </div>  </template> <template x-if="assets_locations_filtered.length == 0" data-astro-cid-2qz5zwg2> <div class="wrapper" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: 0;--wrapper_padding_inline_end: 0;--wrapper_padding_block: 0;--wrapper_max_width: var(--max-text-width);">   <p data-astro-cid-2qz5zwg2>No assets found</p>   </div>   </template>   </div>  "
`;
